specs/consul/consul.spec

```diff
@@ -0,0 +1,263 @@
+
+%description client
+A client is an agent that forwards all RPCs to a server. The client is
+relatively stateless. The only background activity a client performs is taking
+part in the LAN gossip pool. This has a minimal resource overhead and consumes
+only a small amount of network bandwidth.
+
+################################################################################
+
+%package server
+Summary:        Consul Server
+Group:          Application/Internet
+
+Requires:       %{name} = %{version}-%{release}
+
+%description server
+A server is an agent with an expanded set of responsibilities including
+participating in the Raft quorum, maintaining cluster state, responding to RPC
+queries, exchanging WAN gossip with other datacenters, and forwarding queries
+to leaders or remote datacenters.
+
+################################################################################
+
+%prep
+%setup -qn %{name}-%{version}
+
+mkdir -p .src/github.com/hashicorp/%{name}
+mv * .src/github.com/hashicorp/%{name}/
+mv .src src
+
+%build
+export GOPATH=$(pwd)
+export XC_OS=$(go env GOOS)
+export XC_ARCH=$(go env GOARCH)
+export GO15VENDOREXPERIMENT=1
+export CGO_ENABLED=0
+export GIT_IMPORT="github.com/hashicorp/consul/version"
+export GOLDFLAGS="-X $GIT_IMPORT.GitDescribe=%{version}"
+
+pushd src/github.com/hashicorp/%{name}
+  %{__make} %{?_smp_mflags} tools || :
+  $GOPATH/bin/gox -osarch="${XC_OS}/${XC_ARCH}" \
+                  -ldflags "${GOLDFLAGS}" \
+                  -tags="consul" \
+                  -output "$GOPATH/%{name}" .
+popd
+
+%install
+rm -rf %{buildroot}
+
+install -dm 755 %{buildroot}%{_bindir}
+install -dm 755 %{buildroot}%{_sysconfdir}/sysconfig
+install -dm 755 %{buildroot}%{_sysconfdir}/%{name}/server
+install -dm 755 %{buildroot}%{_sysconfdir}/%{name}/client
+install -dm 755 %{buildroot}%{_sharedstatedir}/%{name}
+install -dm 755 %{buildroot}%{_unitdir}
+install -dm 755 %{buildroot}%{_localstatedir}/log/%{name}-server
+install -dm 755 %{buildroot}%{_localstatedir}/log/%{name}-client
+
+install -pm 755 %{name} %{buildroot}%{_bindir}/
+install -pm 644 %{SOURCE1} %{buildroot}%{_sysconfdir}/sysconfig/%{name}-client
+install -pm 644 %{SOURCE2} %{buildroot}%{_sysconfdir}/sysconfig/%{name}-server
+install -pm 644 %{SOURCE3} %{buildroot}%{_unitdir}/
+install -pm 644 %{SOURCE4} %{buildroot}%{_unitdir}/
+install -pm 644 %{SOURCE5} %{buildroot}%{_sysconfdir}/%{name}/client/config.json
+install -pm 644 %{SOURCE6} %{buildroot}%{_sysconfdir}/%{name}/server/config.json
+
+%clean
+rm -rf %{buildroot}
+
+################################################################################
+
+%pre
+getent group %{name} >/dev/null || groupadd -r %{name}
+getent passwd %{name} >/dev/null || \
+    useradd -r -M -g %{name} -d %{_sharedstatedir}/%{name} \
+            -s /sbin/nologin %{name}
+exit 0
+
+%post client
+if [[ $1 -eq 1 ]] ; then
+    %{__systemctl} enable %{name}-client.service &>/dev/null || :
```

Comment:
Currently, we have packages for CentOS 6 and 7. Without init script package for CentOS 6 will be broken.

Reply:
Fixed (0a34c6cfcaf9d82bb710a77d8e28e4c36016bfa3).